### PR TITLE
Add missing relation access record for local utility command

### PR DESCRIPTION
**Pull request by onderkalaci**
While testing 5670dffd338bec21372a75da40938b9494f5edc1, I realized
that we have a missing RecordNonDistTableAccessesForTask() for
local utility commands.

Although we don't have to record the relation access for local
only cases, we really want to keep the behaviour for scale-out
be the same with single node on all aspects. We wouldn't want
any single node complex transaction to work on single machine,
but not on multi node cluster. Hence, we apply the same restrictions.

For example, on a distributed cluster, the following errors, and
after this commit this errors locally as well

```SQL
CREATE TABLE ref(a int primary key);
INSERT INTO ref VALUES (1);

CREATE TABLE dist(a int REFERENCES ref(a));
SELECT create_reference_table('ref');
SELECT create_distributed_table('dist', 'a');

BEGIN;
		SELECT * FROM dist;
		TRUNCATE ref CASCADE;

ERROR:  cannot execute DDL on table "ref" because there was a parallel SELECT access to distributed table "dist" in the same transaction
HINT:  Try re-running the transaction with "SET LOCAL citus.multi_shard_modify_mode TO 'sequential';"

COMMIT;
```

We also add the comprehensive test suite and run the same locally.

### CODE REVIEW

File: src/test/regress/expected/relation_access_tracking_single_node.out [hunk 1]
```diff
@@ -0,0 +1,1090 @@
+---------------------------------------------------------------------
```
Comment (onderkalaci) [inline src/test/regress/expected/relation_access_tracking_single_node.out:1]: this produces the same output with src/test/regress/expected/relation_access_tracking.out

which is what we want

**Review by marcocitus — APPROVED** [review]

File: src/backend/distributed/executor/local_executor.c [hunk 2]
```diff
@@ -413,6 +413,8 @@ ExtractParametersForLocalExecution(ParamListInfo paramListInfo, Oid **parameterT
 static void
 LocallyExecuteUtilityTask(Task *task)
 {
+	RecordNonDistTableAccessesForTask(task);
```
Comment (tejeswarm) [inline src/backend/distributed/executor/local_executor.c:416]: Just add a comment that we are adding for parity with multi node behaviour.

**Review by tejeswarm — APPROVED** [review]

**Review by onurctirtir — COMMENTED** [review]
let me have a quick look now, before merging

File: src/test/regress/expected/relation_access_tracking_single_node.out (hunk 1 above)
Comment (onurctirtir) [reply src/test/regress/expected/relation_access_tracking_single_node.out:1]: Then should we instead add this into arbitrary config tests ?

**Review by onurctirtir — COMMENTED** [review]
Since we call RecordNonDistTableAccessesForTask at LocallyExecuteUtilityTask, not in ExecuteUtilityCommand, what about the external callers of ExecuteUtilityCommand ?

**Comment by onderkalaci** [comment]
> Since we call RecordNonDistTableAccessesForTask at LocallyExecuteUtilityTask, not in ExecuteUtilityCommand, what about the external callers of ExecuteUtilityCommand ?

I'm not sure I fully get the question, but here is my answer. `ExecuteUtilityCommand` is a high-level API, if the command passed to `ExecuteUtilityCommand`.

When the utility command should run in a distributed manner, in the end, gets into adaptive executor, `AssignPlacementListToConnection->RecordRelationAccessIfNonDistTable()`, where we do the trick.

Similarly, when the utility command should run locally, it gets into `LocallyExecuteUtilityTask`-> `RecordNonDistTableAccessesForTask` -> `RecordRelationAccessIfNonDistTable()`

Maybe part of the confusion is that we use `RecordNonDistTableAccessesForTask` for the local case, and `RecordRelationAccessIfNonDistTable` for the distributed case. But, in the end, `RecordNonDistTableAccessesForTask` calls into `RecordRelationAccessIfNonDistTable`. 

We do this, because in the remote executor, we already calculate which placement is accessed and its access mode. Hence, to need a high level function like  `RecordNonDistTableAccessesForTask`. But, maybe we should do for readability.

Does that answer your question?

**Comment by onderkalaci** [comment]
> Then should we instead add this into arbitrary config tests ?

Hmm, let me see if I can do that easily, if so, I can add

**Comment by onderkalaci** [comment]
> Then should we instead add this into arbitrary config tests ?
Hmm, let me see if I can do that easily, if so, I can add


Ah, we need superuser to create C functions, and arbitrary configs run without superuser. I'll merge this now, but will think on how to get these tests to there separately. We might need some `run_superuser` file in arbitrary configs, which should not be in this PR

**Comment by onurctirtir** [comment]
> Maybe part of the confusion is that we use RecordNonDistTableAccessesForTask for the local case, and RecordRelationAccessIfNonDistTable for the distributed case. But, in the end, RecordNonDistTableAccessesForTask calls into RecordRelationAccessIfNonDistTable.

oh, got it now, thanks!

**Review by onurctirtir — APPROVED** [review]

**Pull request merged**